libs/feature/map/src/lib/utils/map-utils-wms.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core'
+import { WmsEndpoint } from '@camptocamp/ogc-client'
+import { MapContextLayerWmsModel } from '../map-context/map-context.model'
+import { ProxyService } from '@geonetwork-ui/util/shared'
+import { from, Observable } from 'rxjs'
+import { map } from 'rxjs/operators'
+import { LONLAT_CRS_CODES } from './projections'
+
+export type WMSEndPoint = any
+export type WMSLayerFull = any
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MapUtilsWMSService {
+  constructor(private proxy: ProxyService) {}
+
+  getCapabilities(layer: MapContextLayerWmsModel): Observable<WMSEndPoint> {
+    return from(new WmsEndpoint(this.proxy.getProxiedUrl(layer.url)).isReady())
+  }
+
+  getLayerFull(layer: MapContextLayerWmsModel): Observable<WMSLayerFull> {
+    return this.getCapabilities(layer).pipe(
+      map((endpoint) => endpoint.getLayerByName(layer.name))
+    )
+  }
+
+  getLayerLonLatBBox(layer: MapContextLayerWmsModel) {
+    return this.getLayerFull(layer).pipe(
+      map((wmsLayerFull) => this.getLonLatBBox(wmsLayerFull))
+    )
+  }
+
+  getLonLatBBox(wmsLayerFull: WMSLayerFull) {
```

Comment:
This utility functions is really nice, and to be honest I think it would fit very well in ogc-client. The philosophy of the library is to save developer from writing this kind of things; for example we do something similar in ogc-client for selecting a mime type that corresponds to geojson for WFS requests.

This can be done later though, I guess

Reply:
Yes I wanted to put it in ogc-client to be honest, but wasn't sure.
Let's do that in a second step.